clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -209,7 +210,10 @@ private SecurityStore maybeCreateNewTruststore(Map<String, ?> configs) {
     SSLContext createSSLContext(SecurityStore keystore, SecurityStore truststore) throws GeneralSecurityException, IOException  {
         SSLContext sslContext;
         if (provider != null)
-            sslContext = SSLContext.getInstance(protocol, provider);
+            if (provider.equals("Conscrypt"))
```

Comment:
performance nit - can have a (final) Supplier<SSLContext> thats created on boot/config and just called here.
would eliminate a string comparison at runtime :-)